tests/model_serving/model_server/kserve/negative/test_tensor_shape_boundary.py

```
@@ -0,0 +1,176 @@
+
+from tests.model_serving.model_server.kserve.negative.utils import (
+    VALID_OVMS_INFERENCE_BODY,
+    assert_pods_healthy,
+    send_inference_request,
+)
+
+pytestmark = pytest.mark.usefixtures("valid_aws_config")
+
+TENSOR_ERROR_EXPECTED_CODES: set[int] = {
+    HTTPStatus.BAD_REQUEST,           # 400
+    HTTPStatus.PRECONDITION_FAILED,   # 412 – OVMS uses this for shape errors
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)